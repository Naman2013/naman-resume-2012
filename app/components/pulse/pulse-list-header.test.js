import React from 'react';
import renderer from 'react-test-renderer';
import jest from 'jest';
import { shallow, mount } from 'enzyme';
import sinon from 'sinon';
import PulseListHeader from './pulse-list-header';

it('it should render a menu', () => {
  const testData = {
    title: '',
    subtitle: '',
    showCreateNewPostButton: false,
  };
  const tree = renderer.create(
    <PulseListHeader
      {...testData}
    />,
  );

  expect(tree.toJSON()).toMatchSnapshot();
});

it('it should render a defined header', () => {
  const testData = {
    title: '',
    subtitle: '',
    showCreateNewPostButton: false,
  };
  const component = shallow(<PulseListHeader {...testData} />);
  expect(component).toBeDefined();
});

it('it should render a title and subtitle if one is passed', () => {
  const testData = {
    title: 'Illuminations',
    subtitle: 'subtitle',
    showCreateNewPostButton: false,
  };
  const component = mount(<PulseListHeader {...testData} />);
  expect(component.find('.header-title').first().text()).toContain(testData.title);
  expect(component.find('.header-subtitle').first().text()).toContain(testData.subtitle);
});
