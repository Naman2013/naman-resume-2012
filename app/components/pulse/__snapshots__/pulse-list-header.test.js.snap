// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render a menu 1`] = `
<div
  className="pulseListHeader"
  data-jsx={674672964}
>
  <div
    className="title"
    data-jsx={674672964}
  >
    <h1
      className="header-title"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      data-jsx={674672964}
    />
    <div
      className="header-subtitle"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      data-jsx={674672964}
    />
  </div>
  <style
    jsx={true}
  >
    
    
          .pulseListHeader {
            display: flex;
            flex-direction: row;
            flex-wrap: wrap;
            justify-content: space-between;
            font-family: "Brandon Grotesque", "brandon-grotesque", sans-serif;
            background: url('https://vega.slooh.com/assets/images/graphics/milkeyway_header.jpg') no-repeat;
            background-size: cover;
            min-height: 145px;
            padding: 50px;
            position: relative;
          }
    
          .title {
    
          }
    
          :global(.header-button) {
            margin: 15px 15px;
          }
    
          .header-title {
            margin: 0;
            color: #fff;
            text-transform: uppercase;
            font-weight: 600;
            font-size: 35px;
          }
    
          .header-subtitle {
            font-size: 1em;
            margin: 0 0 20px;
            padding: 0;
            text-transform: none;
            font-weight: normal;
            color: #fff;
            clear: both;
          }
    
          @media(max-width:375px){
            .header-title {
              font-size: 35px;
            }
    
          }
    
        
  </style>
</div>
`;
